python/ql/src/Security/CWE-327/InsecureDefaultProtocol.ql

```diff
@@ -11,20 +11,19 @@
  */
 
 import python
-
-FunctionValue ssl_wrap_socket() { result = Value::named("ssl.wrap_socket") }
-
-ClassValue ssl_Context_class() { result = Value::named("ssl.SSLContext") }
+import semmle.python.ApiGraphs
 
 CallNode unsafe_call(string method_name) {
-  result = ssl_wrap_socket().getACall() and
+  result = API::moduleImport("ssl").getMember("wrap_socket").getACall().asCfgNode() and
   not exists(result.getArgByName("ssl_version")) and
   method_name = "deprecated method ssl.wrap_socket"
   or
-  result = ssl_Context_class().getACall() and
+  result = API::moduleImport("ssl").getMember("SSLContext").getACall().asCfgNode() and
   not exists(result.getArgByName("protocol")) and
   not exists(result.getArg(0)) and
-  method_name = "ssl.SSLContext"
+  method_name = "ssl.SSLContext" and
+  // in version 3.4, flags were introduced to modify cotexts created with default values
+  (major_version() < 3 or minor_version() < 4)
```

Comment:
Notice that these depend on the Python version used when creating the database, so `(major_version() < 3 or minor_version() < 4)` will effectively be the same as `none()`.

From my point of view, if you want to do special handling of Python 2, that's fine, but let's not specialize to Python 3.3 :smile:

Reply:
Perhaps I misunderstand completely what these predicates do, can you elaborate on how this means `none()`, surely either disjunct is true some of the time? Do you mean that "for Python 3, this is effectively `none()`"?

Reply:
Oh wait, I read that as an `and`... my bad. 

```suggestion
  // in version 3.4, flags were introduced to modify cotexts created with default values
  (major_version() = 2 or major_version() = 3 and minor_version() < 4)
```

I know this effectively means the same, but for me, it reads more easily.

Reply:
That is so...explicit..